### PR TITLE
Adds nfts.getContractEvents and nfts.getCollectionEvents

### PR DESCRIPTION
**Pull request by johnpmitsch**
I left out the filters but going to follow up with adding them in general, it was a little tricky with the enums that apollo-codegen emits since they won't work with strings: [see issue here](https://github.com/apollographql/apollo-tooling/pull/1750)

### CODE REVIEW

File: packages/libs/sdk/src/api/controllers/events.ts
Comment (johnpmitsch) [inline packages/libs/sdk/src/api/controllers/events.ts:1]: I didn't use this controller, but this is set up for some other methods

File: packages/libs/sdk/src/api/api.ts [hunk 1]
```diff
@@ -101,11 +101,8 @@ export class API {
 
     const cacheStructure = new InMemoryCache({
       possibleTypes: generatedPossibleTypes.possibleTypes,
-      typePolicies: {
-        NFT: {
-          keyFields: ['contractAddress', 'tokenId'],
-        },
-      },
+      // TODO: Figure out type policies
+      typePolicies: {},
```
Comment (emanuelpinho) [inline packages/libs/sdk/src/api/api.ts:105]: wasn't the previous working?

**Review by emanuelpinho — APPROVED** [review]
Well structured, just small validation

File: packages/libs/sdk/src/api/api.ts (hunk 1 above)
Comment (johnpmitsch) [reply packages/libs/sdk/src/api/api.ts:105]: I was seeing this:

```
Invariant Violation: Missing field 'contractAddress' while extracting keyFields from {"tokenEvents":{"pageInfo":{"endCursor":"T2Zmc2V0Q29ubmVjdGlvbjox","hasNextPage":false,"hasPreviousPage":false,"startCursor":"T2Zmc2V0Q29ubmVjdGlvbjow","__typename":"PageInfo"},"edges":[{"node":{"blockNumber":13181737,"fromAddress":"0x0000000000000000000000000000000000000000","timestamp":"2021-09-08T00:03:32.000Z","toAddress":"0x5467ba7a07a11769c2bd499f9fbe94a6b0eff269","transactionHash":"0xf78f4c8ee2029cf961a3cad6c6cafab3fbe14eb1932d705d655cba8b6b071265","transferIndex":398,"type":"MINT","tokenQuantity":1,"__typename":"TokenMintEvent"},"__typename":"NFTTokenEventsConnectionEdge"},{"node":{"blockNumber":14604458,"fromAddress":"0x5467ba7a07a11769c2bd499f9fbe94a6b0eff269","timestamp":"2022-04-17T18:50:08.000Z","toAddress":"0xa673b62171ed57d184214aae4bc0746024ab8cd8","transactionHash":"0xa69713bc3b43b90f4687ca8fe853021b828017feed6104dd4394413800fe85cc","transferIndex":576,"type":"SALE","marketplace":"OPENSEA","receivedTokenContractAddress":"0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee","receivedTokenId":null,"sentTokenId":2000,"__typename":"TokenSaleEvent"},"__typename":"NFTTokenEventsConnectionEdge"}],"__typename":"NFTTokenEventsConnection"},"__typename":"ERC721NFT"}
```

My thought is to push the type policies in general out until later, especially as we add more queries

**Pull request merged**